### PR TITLE
RGW - Zipper - Pass invaliate down from RadosStore

### PR DESCRIPTION
**Pull request by dang**
When we invalidate the state of a RadosObject, we need to also
invalidate it in the ctx for that object.

Signed-off-by: Daniel Gryniewicz <dang@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [x] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/rgw/rgw_sal_rados.h [hunk 1]
```diff
@@ -383,6 +383,10 @@ class RadosObject : public StoreObject {
 
     virtual ~RadosObject();
 
+    virtual void invalidate() override {
+      StoreObject::invalidate();
+      rados_ctx->invalidate(get_obj());
```
Comment (cbodley) [inline src/rgw/rgw_sal_rados.h:388]: i see that `RadosObject` has a default ctor that leaves this `rados_ctx` uninitialized. should we initialize it to `nullptr` there and add a null check here? or maybe drop the default ctor?

**Review by cbodley — COMMENTED** [review]
👍

**Review by cbodley — APPROVED** [review]
(that 👍 was supposed to be an approval)

File: src/rgw/rgw_sal_rados.h (hunk 1 above)
Comment (dang) [reply src/rgw/rgw_sal_rados.h:388]: Good question.  I'll look at dropping it.

**Comment by dang** [comment]
https://pulpito.ceph.com/dang-2022-09-01_10:01:07-rgw-wip-dang-zipper-invalidate-distro-default-smithi/

Only known failures

**Pull request merged**